### PR TITLE
authz: remove upsert for updating user pending permissions

### PR DESCRIPTION
**Pull request by unknwon**
We used to rely on `ON CONFLICT ON CONSTRAINT` to do upsert when we do batch update on `user_pending_permissions` table. However, there are reasons why the approach in this PR is better (use update not upsert):

1. The primary key (`id`) of `user_pending_permissions` table is increased rapidly every time hits `ON CONFLICT ON CONSTRAINT` for no obvious benefit. (Have seen a customer with few thousand rows but `MAX(id)` >80,000,000).
2. At the moment of "upsert", we already have `id`s on hand, could have just used it to directly update rows with these `id`s as they are primary keys.

This PR does not solve 2. entirely but at least slows down `id` increment pace by half. Another half is caused by:

https://github.com/sourcegraph/sourcegraph/blob/e0145db0159fad19a1f48735a27986e502cbf3c9/enterprise/internal/db/perms_store.go#L598-L599

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Talked about this in a Zoom chat, seems to be valid

**Review by daxmc99 — APPROVED** [review]
LGTM! Reviewed over Zoom

File: enterprise/internal/db/perms_store.go [hunk 1]
```diff
@@ -871,12 +855,8 @@ DO UPDATE SET
 			return nil, ErrPermsUpdatedAtNotSet
 		}
 
-		items[i] = sqlf.Sprintf("(%s, %s, %s, %s, %s, %s, %s)",
-			ps[i].ServiceType,
-			ps[i].ServiceID,
-			ps[i].BindID,
-			ps[i].Perm.String(),
-			ps[i].Type,
+		items[i] = sqlf.Sprintf("(%s::INT, %s::BYTEA, %s::TIMESTAMP WITH TIME ZONE)",
```
Comment (daxmc99) [inline enterprise/internal/db/perms_store.go:815]: TIl about Postgres types!

**Comment by unknwon** [comment]
Thanks again @eseliger and @daxmc99 for pair reviewing with me! Merging 🎉

**Pull request merged**